### ISSUE TITLE
PHP tag must be on a line by itself - creates invalid output in HTML attributes

### ISSUE THREAD

**Issue by kkmuffme** [PHPCSStandards/PHP_CodeSniffer#829]
## Describe the bug

>Opening PHP tag must be on a line by itself
>Closing PHP tag must be on a line by itself

### Code sample
```php
<span data-foo="<?php
echo "hello";
?>">
```

### Custom ruleset
https://github.com/WordPress/WordPress-Coding-Standards

### To reproduce
See WP CI in https://github.com/WordPress/wordpress-develop/pull/8335 https://github.com/WordPress/wordpress-develop/actions/runs/13371451585/job/37340655741?pr=8335

## Expected behavior

Putting the PHP opening/closing tags on a separate line when used in attributes means that the attribute will include leading/trailing whitespace in output.
Since browsers know this is a common bug, they tend to ignore it, but it means that e.g. href= includes an invalid URL. 
For some attributes it might actually break it or have unintended behavior e.g. pattern=, placeholder=
Besides the fact, that newlines aren't allowed in attributes.

## Versions (please complete the following information)

I guess 3.11.3 in WP CI

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@kkmuffme This is not reproducible with the information provided. The code sample isn't valid PHP and there is no information on the ruleset (and no, I'm not going to search for it). Please submit self-contained reproducible bug reports, otherwise this issue will be closed.

**Comment by jrfnl** [comment]
@kkmuffme Please do not make any assumption about who will triage an issue. I may be aware of the WordPress Coding Standards, others may not, so please always provide a complete reproduction case to allow anyone who comes across the issue to respond to it.

> If you mean which rule ID specifically causes the error - I don't know, since the WP CI doesn't tell me which rule that is, but just the error message. Perhaps something that could be improved in the WP CI?

I'm not interested in what WP CI does or doesn't do. This is the PHPCS repo, not the WP repo.
Using the `-s` on the command-line will show which sniff is involved and will save whomever comes across the issue from searching. Verifying this ahead of opening an issue will also prevent opening an issue in the wrong repo.

**Comment by jrfnl** [comment]
> Putting the PHP opening/closing tags on a separate line when used in attributes means that the attribute will include leading/trailing whitespace in output.

As for the issue itself. IMO this is a problem of code design and not something which needs to be addressed in PHPCS.

Clearly WordPressCS has decided to enforce multi-line PHP snippets to be formatted with the open and close tags on their own line. That just means you need to make sure your code complies with that.

This can be done in multiple ways, like:
```php
<!-- Use a single-line snippet instead if the code is single-statement. -->
<span data-foo="<?php echo "hello"; ?>">
```
...or ...
```php
<?php
// Execute the logic ahead of sending the output if the code is multi-statement.
$var = 'foo';
if ($something) {
    $var = 'bar';
}
?>
<span data-foo="<?php echo $var; ?>">
```

In other words: this is not a bug, but a support request and doesn't even belong in the issue tracker. It should have been a discussion. Please abide by the guidelines in this repo.

**Comment by jrfnl** [comment]
> Definitely, but I am not the maintainer of the WP code style.

Anything to do with the WP code style rules, should be discussed in their repo, not here.

> > In other words: this is not a bug
> 
> PHPCS suggests (or in this case I think with phpcbf even auto-fixes it) a change, that will result in invalid HTML. How is this not a bug?

How is this invalid HTML ? https://3v4l.org/kGvvM

**Comment by jrfnl** [comment]
> The rule is `PSR12.Files.OpenTag.NotAlone` (and CloseTag accordingly), which is part of this repo, isn't it?

That sniff is part of this repo, but would **not** be triggered by any of the code samples posted here.


> ... but phpcs reports an error for it phpcs suggests to change it to: https://3v4l.org/Yp88k (which technically is also valid. If instead of the data- a id, pattern, src, href,... is used it wouldn't be valid)

Even when using the `WordPress` standard, I cannot reproduce this. Running `phpcs` only reports a missing file comment and running `phpcbf` reports "No fixable errors".

> 
> To avoid pedantery, here's a new example: Code: (valid HTML, phpcs error)
> 
> ```
> <input type="text" pattern="<php
> echo 'abc';
> ?>">
> ```

This code contains a code error and phpcs would report "No PHP code found in the file" (as the PHP open tag is not a PHP open tag). And even once I fix that error, the same happens as before:

> Running `phpcs` only reports a missing file comment and running `phpcbf` reports "No fixable errors".

In other words: you still haven't provided a workable reproduction sample.
Your point about `phpcbf` producing invalid HTML cannot be reproduced. And even if it could, I maintain this is a code design issue.
